### ISSUE TITLE
Classification Loss during Adversarial Training

### ISSUE THREAD

**Issue by jeffz0** [xialeiliu/GFR-IL#5]
Hi Xialei, 

I noticed that in lines 323 and 366 of train.py the classification losses d_loss_lbls and g_loss_lbls are not used in the loss for either the discriminator or the generator. Does this mean only an adversarial loss is enough to generate feature vectors that work for classification?

Thanks a lot,

Jeff

**Comment by xialeiliu** [comment]
Hi Jeff,

Thanks for you question.

The short answer is no, the label information is embedded into the discriminator in a different way, as seen in:
https://github.com/xialeiliu/GFR-IL/blob/master/models/resnet.py#L91-L96

Sorry, we should have removed these not-used lines.

Best,
Xialei

**Comment by jeffz0** [comment]
Hi Xialei,

Thanks for the reply and I see, that makes sense! Did you find this to work better than softmax cross-entropy for the classification loss (d_loss_lbls)? 

Best,

Jeff

**Comment by xialeiliu** [comment]
Yes, both should work ok, we found in our experiments the one we used in the end gives slightly better performance.  

Best,
Xialei

**Comment by jeffz0** [comment]
Awesome, thanks again for the prompt responses. Greatly appreciate it.
